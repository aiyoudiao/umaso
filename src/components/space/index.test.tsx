import type { ReactTestRendererJSON } from 'react-test-renderer'
import { describe, it } from 'vitest'
import * as React from 'react'
import renderer from 'react-test-renderer'
import Tag from '../tag/index'
import Space from './index'

describe('测试标签', () => {
  it('render Tag ', () => {
    const rest = renderer.create(
      <div>
        <Space>
          <Tag>标签 0</Tag>
          <Tag size="mini">标签 1</Tag>
          <Tag size="small">标签 2</Tag>
          <Tag size="medium">标签 3</Tag>
        </Space>
      </div>,
    )
    expect(rest).toMatchSnapshot()
  })
  it('Space direction', () => {
    const rest = renderer.create(
      <Space direction="vertical">
        <Tag>标签 0</Tag>
        <Tag size="mini">标签 1</Tag>
        <Tag size="small">标签 2</Tag>
        <Tag size="medium">标签 3</Tag>
      </Space>,
    )

    const json = rest.toJSON() as ReactTestRendererJSON
    const className = json.props.className
    expect(className).toContain('el-space--vertical')
  })
  it('Space size', () => {
    const rest = renderer.create(
      <Space size={60}>
        <Tag>标签 0</Tag>
        <Tag size="mini">标签 1</Tag>
        <Tag size="small">标签 2</Tag>
        <Tag size="medium">标签 3</Tag>
      </Space>,
    )
    const json = rest.toJSON() as ReactTestRendererJSON
    expect(json.props.style.gap).toEqual('60px')
  })
  it('Space align', () => {
    const rest = renderer.create(
      <Space align="center">
        <Tag>标签 0</Tag>
        <Tag size="mini">标签 1</Tag>
        <Tag size="small">标签 2</Tag>
        <Tag size="medium">标签 3</Tag>
      </Space>,
    )

    const json = rest.toJSON() as ReactTestRendererJSON
    const className = json.props.className
    expect(className).toContain('el-space--align-center')
  })
  it('Space wrap', () => {
    const rest = renderer.create(
      <Space wrap>
        <Tag>标签 0</Tag>
        <Tag size="mini">标签 1</Tag>
        <Tag size="small">标签 2</Tag>
        <Tag size="medium">标签 3</Tag>
      </Space>,
    )
    const json = rest.toJSON() as ReactTestRendererJSON
    expect(json.props.style.flexWrap).toEqual('wrap')
  })
  it('Space split', () => {
    const rest = renderer.create(
      <Space split="|">
        <Tag>标签 0</Tag>
        <Tag size="mini">标签 1</Tag>
        <Tag size="small">标签 2</Tag>
        <Tag size="medium">标签 3</Tag>
      </Space>,
    )

    const json = rest.toJSON() as ReactTestRendererJSON
    const children = json.children
    expect(children?.length).toEqual(7)
  })
})
