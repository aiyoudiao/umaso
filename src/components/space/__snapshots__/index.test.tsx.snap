// Vitest Snapshot v1

exports[`测试标签 > render Space  1`] = `
<div>
  <div
    className="el-space el-space--horizontal el-space--align-start"
    style={
      {
        "flexWrap": "nowrap",
        "gap": "8px",
      }
    }
  >
    <span
      className="el-tag--light el-tag"
    >
      标签 0
    </span>
    <span
      className="el-tag--mini el-tag--light el-tag"
    >
      标签 1
    </span>
    <span
      className="el-tag--small el-tag--light el-tag"
    >
      标签 2
    </span>
    <span
      className="el-tag--medium el-tag--light el-tag"
    >
      标签 3
    </span>
  </div>
</div>
`;

exports[`测试标签 > render Tag  1`] = `
<div>
  <div
    style={
      {
        "gap": undefined,
      }
    }
  />
</div>
`;
